compiler_opt/es/regalloc_trace/regalloc_trace_worker.py

```diff
@@ -0,0 +1,153 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Worker for regalloc using trace-based cost modeling."""
+
+from typing import Optional, Collection
+import os
+import pathlib
+import subprocess
+import json
+import concurrent.futures
+import tempfile
+
+import gin
+
+from compiler_opt.rl import corpus
+from compiler_opt.distributed import worker
+from compiler_opt.rl import policy_saver
+
+
+@gin.configurable
+class RegallocTraceWorker(worker.Worker):
+  """A worker that produces rewards for a given regalloc policy."""
+
+  def __init__(self, clang_path: str, basic_block_trace_model_path: str,
+               thread_count: int, corpus_path: str):
+    self._clang_path = clang_path
+    self._basic_block_trace_model_path = basic_block_trace_model_path
```

Comment:
explain somewhere what this is. Also what it expects its inputs to be. I realize this is upcoming, but better have more information local to where the reader is than less.